burn-dataset/src/dataset/sqlite.rs

```diff
@@ -218,4 +439,217 @@ mod tests {
+            SqliteDatasetSaver::<Complex>::new("preprocessed")
+                .with_base_dir(temp_dir_str.as_str())
+                .with_splits(&["train", "test"])
+                .init(),
+            tmp_dir,
+        )
+    }
+
+    #[test]
+    pub fn sqlite_dataset_saver_new() {
+        let random_name = format!("dataset-{}", rand::random::<u32>());
+        let ds_saver = SqliteDatasetSaver::<Complex>::new(random_name.as_str());
+
+        assert_eq!(ds_saver.name, random_name.as_str());
+        assert_eq!(ds_saver.base_dir, None);
+        assert!(!ds_saver.overwrite);
+        assert!(!ds_saver.is_initialized);
+        assert_eq!(ds_saver.splits, None);
+        assert!(ds_saver.conn_pool.is_none());
+        assert!(ds_saver
+            .db_file()
+            .ends_with(format!("{}.db", random_name).as_str()));
+        assert!(
+            !ds_saver.exists(),
+            "db file {} should not exist (remove it manually if it does)",
+            ds_saver.db_file()
+        );
+    }
+
+    #[rstest]
+    pub fn sqlite_dataset_saver_init(tmp_dir: TempDir) {
+        let temp_dir_str = tmp_dir.path().to_str().unwrap().to_string();
+
+        let ds_saver = SqlDsSaver::new("preprocessed")
+            .with_base_dir(temp_dir_str.as_str())
+            .with_splits(&["train", "test"])
+            .init();
+
+        assert_eq!(ds_saver.base_dir, Some(temp_dir_str));
+
+        assert!(ds_saver.is_initialized);
+        assert_eq!(
+            ds_saver.splits,
+            Some(vec!["train".to_string(), "test".to_string()])
+        );
+        assert!(ds_saver.conn_pool.is_some());
+        assert!(ds_saver.exists());
+    }
+
+    #[rstest]
+    #[should_panic]
+    pub fn sqlite_dataset_saver_init_existing(tmp_dir: TempDir) {
+        let temp_dir_str = tmp_dir.path().to_str().unwrap().to_string();
+
+        // Create a file in the temp dir
+        let _ds_saver1 = SqlDsSaver::new("preprocessed")
+            .with_base_dir(temp_dir_str.as_str())
+            .with_splits(&["train", "test"])
+            .init();
+
+        // Again create a file in the temp dir and it should fail
+        let _ds_saver2 = SqlDsSaver::new("preprocessed")
+            .with_base_dir(temp_dir_str.as_str())
+            .with_splits(&["train", "test"])
+            .init();
+    }
+
+    #[rstest]
+    pub fn sqlite_dataset_saver_init_with_overwrite(tmp_dir: TempDir) {
+        let temp_dir_str = tmp_dir.path().to_str().unwrap().to_string();
+
+        // Create a file in the temp dir
+        let _ds_saver1 = SqlDsSaver::new("preprocessed")
+            .with_base_dir(temp_dir_str.as_str())
+            .with_splits(&["train", "test"])
+            .init();
+
+        // Again create a file in the temp dir and it should fail
+        let _ds_saver2 = SqlDsSaver::new("preprocessed")
+            .with_base_dir(temp_dir_str.as_str())
+            .with_splits(&["train", "test"])
+            .with_overwrite(true)
+            .init();
+    }
```

Comment:
It seems like testing privates 🤔. Those functions seem like utilities more than a public APIs that are going to survive through time.

Reply:
The only correct way to verify if something is initialized is to check it's state. We are checking the private fields but this is unit testing so we are allowed in this particular case. These fields are used in subsequent calls and it's important they're set correctly. Alternatively I could make these fields public or create public methods, also no harm but I didn't see use cases.


The line 545 should say:         // Again create a file in the temp dir and it should NOT fail